### PR TITLE
Add the fail2ban recipe

### PR DESCRIPTION
**Pull request by radamanthus**

### CODE REVIEW

File: cookbooks/fail2ban/attributes/default.rb [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+  # gamin:   requires Gamin (a file alteration monitor) to be installed. If Gamin
+  #          is not installed, Fail2ban will use polling.
+  # polling: uses a polling algorithm which does not require external libraries.
+  # auto:    will choose Gamin if available and polling otherwise.
+  'backend'     => 'auto',
+  # mail configuration
+  'mail'      => {
+    'destination'   => 'security_control@company.com',
+    'sender'        => 'root@localhost'
+  },
```
Comment (dvalfre) [inline cookbooks/fail2ban/attributes/default.rb]: for email alerting to work the 'ssmtp' recipe has to be enabled.  Worth adding a note on the Readme.

File: custom-cookbooks/fail2ban/cookbooks/custom-fail2ban/attributes/default.rb [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+  # gamin:   requires Gamin (a file alteration monitor) to be installed. If Gamin
+  #          is not installed, Fail2ban will use polling.
+  # polling: uses a polling algorithm which does not require external libraries.
+  # auto:    will choose Gamin if available and polling otherwise.
+  'backend'     => 'auto',
+  # mail configuration
+  'mail'      => {
+    'destination'   => 'security_control@company.com',
+    'sender'        => 'root@localhost'
+  },
```
Comment (dvalfre) [inline custom-cookbooks/fail2ban/cookbooks/custom-fail2ban/attributes/default.rb]: for email alerting to work the 'ssmtp' recipe has to be enabled. Worth adding a note on the Readme.

**Comment by radamanthus** [comment]
@dvalfre I added a Dependencies section on the README to mention that the ssmtp recipe is needed. Mind taking another look?

**Comment by dvalfre** [comment]
👍

**Pull request merged**